airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -125,12 +125,13 @@ def _render_log_id(self, ti: TaskInstance, try_number: int) -> str:
             else:
                 log_id_template = self.log_id_template
 
-        dag = ti.task.dag
-        assert dag is not None  # For Mypy.
         try:
-            data_interval: Tuple[datetime, datetime] = dag.get_run_data_interval(dag_run)
+            dag = ti.task.dag
         except AttributeError:  # ti.task is not always set.
             data_interval = (dag_run.data_interval_start, dag_run.data_interval_end)
+        else:
+            assert dag is not None  # For Mypy.
+            data_interval: Tuple[datetime, datetime] = dag.get_run_data_interval(dag_run)
```

Comment:
```suggestion
            data_interval = dag.get_run_data_interval(dag_run)
```

I _think_ this can be removed now?